sound/soc/sof/sof-client.h

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+ */
+
+#ifndef __SOUND_SOC_SOF_CLIENT_H
+#define __SOUND_SOC_SOF_CLIENT_H
+
+#include <linux/completion.h>
+#include <linux/device.h>
+#include <linux/device/driver.h>
+#include <linux/kernel.h>
+#include <linux/list.h>
+#include <linux/virtual_bus.h>
+#include "sof-priv.h"
+
+#define SOF_CLIENT_PROBE_TIMEOUT_MS 2000
+
+enum sof_client_type {
+	SOF_CLIENT_AUDIO,
+	SOF_CLIENT_IPC,
+};
+
+/* SOF client device */
+struct sof_client_dev {
+	struct virtbus_device vdev;
+	struct snd_sof_dev *sdev;
+	struct list_head list;	/* item in SOF core client drv list */
+	struct completion probe_complete;
+	void *data;
+};
+
+struct sof_client_drv {
+	const char *name;
+	enum sof_client_type type;
+	const struct sof_client_ops *ops;
+	struct virtbus_driver virtbus_drv;
+};
+
+/* client-specific ops, all optional */
+struct sof_client_ops {
+	/* check if client is requesting DSP to remain in D0 during suspend */
+	bool (*pm_remain_in_D0)(struct sof_client_dev *cdev);
+};
+
+#define virtbus_dev_to_sof_client_dev(virtbus_dev) \
+	container_of(virtbus_dev, struct sof_client_dev, vdev)
+
+static inline int sof_client_drv_register(struct sof_client_drv *drv)
+{
+	return virtbus_register_driver(&drv->virtbus_drv);
+}
+
+static inline void sof_client_drv_unregister(struct sof_client_drv *drv)
+{
+	virtbus_unregister_driver(&drv->virtbus_drv);
+}
```

Comment:
this could be construed as re-wrapping a GPL symbol with a BSD one. Maybe put in the same file as sof-client.c or change the license as GPL here as well?

Reply:
done

Reply:
Interesting, would it really be one? This is an inline function, so, the build-link system "should" make sure, that where it is used the licence is compatible, right? Re-wrapping to me would mean, if we `EXPORT_SYMBOL()` a function, that wraps a GPL-only exported function? I don't remember what the previous version was like: @ranj063 by "done" you mean you changed the licence in this *header* to GPL or that you changed `MODULE_LICENSE()` of `sof-client.c`? If the former - I don't think it matters for symbol resolution, if the latter - this header will also be included in other `.c` files, right?

Reply:
@lyakh it felt a little weird keeping sof-client.h dual licensed while sof-client.c is only GPL. So I've not changed both of them to GPL only